dlt/destinations/impl/databricks/databricks.py

```diff
@@ -54,122 +53,221 @@ def run(self) -> None:
+        if is_local_file:
+            # local file by uploading to a temporary volume on Databricks
+            from_clause, file_name = self._handle_local_file_upload(self._file_path)
+            credentials_clause = ""
+            orig_bucket_path = None  # not used for local file
+        else:
+            # staged file
+            from_clause, credentials_clause, file_name, orig_bucket_path = (
+                self._handle_staged_file()
+            )
+
+        # decide on source format, file_name will either be a local file or a bucket path
+        source_format, format_options_clause, skip_load = self._determine_source_format(
+            file_name, orig_bucket_path
+        )
+
+        if skip_load:
+            # If the file is empty or otherwise un-loadable, exit early
+            return
+
+        statement = self._build_copy_into_statement(
+            qualified_table_name,
+            from_clause,
+            credentials_clause,
+            source_format,
+            format_options_clause,
+        )
+
+        self._sql_client.execute_sql(statement)
+
+    def _handle_local_file_upload(self, local_file_path: str) -> tuple[str, str]:
+        from databricks.sdk import WorkspaceClient
+        import time
+        import io
+
+        w: WorkspaceClient
+
+        credentials = self._job_client.config.credentials
+        if credentials.client_id and credentials.client_secret:
+            # oauth authentication
+            w = WorkspaceClient(
+                host=credentials.server_hostname,
+                client_id=credentials.client_id,
+                client_secret=credentials.client_secret,
+            )
+        elif credentials.access_token:
+            # token authentication
+            w = WorkspaceClient(
+                host=credentials.server_hostname,
+                token=credentials.access_token,
+            )
+
+        file_name = FileStorage.get_file_name_from_file_path(local_file_path)
+        volume_file_name = file_name
+        if file_name.startswith(("_", ".")):
+            volume_file_name = (
+                "valid" + file_name
+            )  # databricks loading fails when file_name starts with - or .
+
+        volume_catalog = self._sql_client.database_name
+        volume_database = self._sql_client.dataset_name
+        volume_name = "_dlt_staging_load_volume"
+
+        # create staging volume name
+        fully_qualified_volume_name = f"{volume_catalog}.{volume_database}.{volume_name}"
+        if self._job_client.config.staging_volume_name:
+            fully_qualified_volume_name = self._job_client.config.staging_volume_name
+            volume_catalog, volume_database, volume_name = fully_qualified_volume_name.split(".")
+
+        self._sql_client.execute_sql(f"""
+            CREATE VOLUME IF NOT EXISTS {fully_qualified_volume_name}
+        """)
+
+        volume_path = f"/Volumes/{volume_catalog}/{volume_database}/{volume_name}/{time.time_ns()}"
```

Comment:
maybe you should take `uniq_id()` from dlt common instead of time? for example on windows you get timer resolution of 20ms. you may get clashing names

Reply:
done